build_scripts/build_usd.py

```diff
@@ -985,13 +985,20 @@ def InstallTIFF(context, force, buildArgs):
 
 def InstallPNG(context, force, buildArgs):
     macArgs = []
-    if MacOS() and Arm():
-        # ensure libpng's build doesn't erroneously activate inappropriate
-        # Neon extensions
-        macArgs = ["-DPNG_HARDWARE_OPTIMIZATIONS=OFF", 
-                   "-DPNG_ARM_NEON=off"] # case is significant
     with CurrentWorkingDirectory(DownloadURL(PNG_URL, context, force)):
-        RunCMake(context, force, buildArgs + macArgs)
+        if MacOS():
+            if Arm():
+                # ensure libpng's build doesn't erroneously activate inappropriate
+                # Neon extensions
+                macArgs = ["-DPNG_HARDWARE_OPTIMIZATIONS=OFF",
+                          "-DPNG_ARM_NEON=off"] # case is significant
+            PatchFile("CMakeLists.txt",
+                [('add_custom_target(gensym DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/libpng.sym")',
+                  'add_custom_target(gensym DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/libpng.sym" genvers)'),
+                 ("add_custom_target(genfiles DEPENDS",
+                  "add_custom_target(genfiles DEPENDS gensym symbol-check")])
```

Comment:
Could a short comment be added here about why this patch is necessary? The build script tries to document why a patch is needed and what software versions are affected so we can understand them during future changes.

Something along the lines of, "This patch is needed to workaround a build issue on MacOS in libpng described at https://github.com/glennrp/libpng/issues/344. This was observed on CMake version blah / Xcode version blah" would be great. (Obviously feel free to add more information).